src/client/gui/client_gui.cpp

```diff
@@ -35,10 +37,25 @@ mp::ClientGui::ClientGui(ClientConfig& config)
 
 int mp::ClientGui::run(const QStringList& arguments)
 {
+    auto ret = 0;
     ArgParser parser;
 
-    auto logger = std::make_shared<mpl::StandardLogger>(mpl::level_from(parser.verbosityLevel()));
-    mpl::set_logger(logger);
+    QCommandLineOption autostart{"exit-unless-autostart",
+                                 "Exit right away, not actually creating a GUI, unless configured to autostart. "
+                                 "Pass this option when auto-starting to honor the autostart setting."};
```

Comment:
Yes, I did that already here: https://github.com/CanonicalLtd/multipass/compare/master...ricab:config-gui-autostart

I just did not push to this repo yet because of the hook issues I mentioned (I've a push check for `//TODO @ricab` tags and it relies on them not being in the base... nevermind).

Reply:
Can you change the help text to say something like "Tell the GUI to apply whatever behaviour is applicable for autostarting." rather than explicitly saying what it is today?